service/controller/app/v1/resource/chartoperator/create.go

```diff
@@ -68,6 +68,22 @@ func (r Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			return microerror.Mask(err)
 		} else {
 			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found chart-operator release %#q", release))
+
+			releaseContent, err := cc.HelmClient.GetReleaseContent(ctx, release)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			if releaseContent.Status == "FAILED" {
+				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("chart-operator release %#q failed to install", release))
+				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating a release %#q", release))
+				err = r.installChartOperator(ctx, cr, true)
```

Comment:
I don't really like that this is `r.installChartOperator` which it's actually an update.

Most of the code is merging the values. That could be extracted to `r.mergeChartOperatorValues` and we could have a separate `r.updateChartOperator` without the need for the boolean flag.

Reply:
Thanks, I moved a bit to new functions.